### PR TITLE
Feature/post endpoint

### PR DESCRIPTION
**Pull request by jitendra2329**
I have added some fields under the POST endpoint. kindly have a look at the same.

### CODE REVIEW

File: scripts/delivery.json [hunk 1]
```diff
@@ -63,8 +63,33 @@
             "401": { "type": "unit" },
             "404": { "type": "unit" }
           }
+        },
+        {
+          "method": "POST",
+          "path": "/:id",
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: We do not doa POST by id, so we can get rid of this path from here.

File: scripts/delivery.json [hunk 2]
```diff
@@ -63,8 +63,33 @@
...
+          "request body": {
+	    "type": "delivery_form",
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: where is this `delivery_form` defined?

File: scripts/delivery.json [hunk 3]
```diff
@@ -63,8 +63,33 @@
...
+	    "schema": {
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: what are these used for?

File: scripts/delivery.json [hunk 4]
```diff
@@ -63,8 +63,33 @@
...
+	      "properties": {
+		"name": { "type": "string" },
+		"data": { "type": "object" }
+	      },
+	      "required": ["name"]
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: Why is this needed?

File: scripts/delivery.json [hunk 5]
```diff
@@ -63,8 +63,33 @@
...
+	    }
+	  },
+	  "security": {
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: Let's not add any security for now. We can add it later if needed.

File: scripts/delivery.json [hunk 6]
```diff
@@ -63,8 +63,33 @@
...
+	    "authentication": "API Key",
+	    "authorization": "Role-based"
+	  },
+          "responses": {
+            "200": { "type": "delivery" },
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "400": { "type": "error"}
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: Let's use 422 instead

File: scripts/delivery.json (hunk 2 above)
Comment (jitendra2329) [reply scripts/delivery.json]: then what would be the path?

File: scripts/delivery.json [hunk 7]
```diff
@@ -48,6 +48,14 @@
         { "name": "mobile_number", "type": "string", "required": false },
         { "name": "email_id", "type": "string", "required": false }
       ]
+    },
+    
+    "delivery_form" : {
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json:53]: This should look similar to delivery model with all the fields except `id` and `estimated_delivery_date`

File: scripts/delivery.json (hunk 2 above)
Comment (Bhavya-verma-knoldus) [reply scripts/delivery.json]: There is a parent path defined [here](https://github.com/Bhavya-verma-knoldus/api/pull/1/files#diff-92fb1c71351e25cd5813a0379d1fad78a6f3a9c38e1b980f21d006bcc2d26247R64). That will be used.

File: scripts/delivery.json [hunk 8]
```diff
@@ -63,8 +80,21 @@
...
+        {
+          "method": "POST",
+          "body": {
+	    "type": "delivery_form"
+	  },
+          "responses": {
+            "200": { "type": "delivery" },
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "422":{ "type": "[error]" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: why is this a sequence of errors?

Comment (jitendra2329) [reply scripts/delivery.json]: maine apibuilder.io se reference liya tha, so I thought ki aise hi dena hoga.
Will change it to single error.

Comment (Bhavya-verma-knoldus) [reply scripts/delivery.json]: Why is this not updated yet?

Comment (jitendra2329) [reply scripts/delivery.json]: local pr kr diya tha, push krna bhul gya tha. sorry for that.

File: scripts/delivery.json [hunk 9]
```diff
@@ -63,8 +80,48 @@
...
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "422":{ "type": "error" }
+          }
+        },
+        
+        {
+          "method": "DELETE",
+          "body": {
+	    "type": "unit"
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: You can skip using the entire "body" field.

File: target/scala-2.12/update/update_cache_2.12/inputs [hunk 10]
```diff
@@ -0,0 +1 @@
+33095130
```
Comment (Bhavya-verma-knoldus) [inline target/scala-2.12/update/update_cache_2.12/inputs]: Remove target folder via `.gitignore`

File: scripts/delivery.json [hunk 11]
```diff
@@ -48,9 +48,26 @@
...
+        { "name": "origin", "type": "address" },
+        { "name": "destination", "type": "address" },
+        { "name": "contact_info", "type": "contact" }	
+    	]
+    },
+    
+    "error": {
+      "fields": [
+        { "name": "code", "type": "string" },
+        { "name": "message", "type": "string" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: Let's make this a sequence of string instead.

File: .gitignore [hunk 12]
```diff
@@ -0,0 +1,4 @@
+/.idea
+/.bsp
+/target
```
Comment (Bhavya-verma-knoldus) [inline .gitignore]: I think this should be `target/`

File: target/scala-2.12/update/update_cache_2.12/inputs (hunk 10 above)
Comment (Bhavya-verma-knoldus) [reply target/scala-2.12/update/update_cache_2.12/inputs]: You also have to delete all the target folders on your local and then push the changes so that all the files from target folder does not exists in this PR.

File: .gitignore (hunk 12 above)
Comment (jitendra2329) [reply .gitignore]: /target will skip the whole folder i think

Comment (jitendra2329) [reply .gitignore]: target/ will skip the content inside the target.
should I change /target to target/ ?

Comment (Bhavya-verma-knoldus) [reply .gitignore]: The intention is to skip the entire target folder. Use whichever line ignore the entire folder.

File: scripts/delivery.json [hunk 13]
```diff
@@ -48,9 +48,26 @@
...
+        { "name": "merchant_id", "type": "string" },
+        { "name": "origin", "type": "address" },
+        { "name": "destination", "type": "address" },
+        { "name": "contact_info", "type": "contact" }	
+    	]
+    },
+    
+    "error": {
+      "fields": [
+        { "name": "code", "type": "[string]" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]: the code would be a string and the messages will be sequence

File: target/scala-2.12/update/update_cache_2.12/inputs (hunk 10 above)
Comment (Bhavya-verma-knoldus) [reply target/scala-2.12/update/update_cache_2.12/inputs]: @jitendra2329 You need to do the above mentioned step so that the files changed in this PR is only 2-3 and not 30+.

File: scripts/delivery.json (hunk 13 above)
Comment (jitendra2329) [reply scripts/delivery.json]: I have changed the same.

File: scripts/delivery.json [hunk 14]
```diff
@@ -63,8 +80,45 @@
...
+          }
+        },
+        
+        {
+          "method": "DELETE",
+          "responses": {
+            "200": { "type": "unit" },
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "422":{ "type": "error" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json]:
```suggestion
            "422": { "type": "error" }
```

File: scripts/delivery.json [hunk 15]
```diff
@@ -63,8 +80,45 @@
...
+        {
+          "method": "PUT",
+          "body": {
+	    "type": "delivery_form"
+	  },
+          "responses": {
+            "200": { "type": "delivery" },
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "422":{ "type": "error" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json:107]:
```suggestion
            "422": { "type": "error" }
```

File: scripts/delivery.json [hunk 16]
```diff
@@ -63,8 +80,45 @@
...
+        {
+          "method": "POST",
+          "body": {
+	    "type": "delivery_form"
+	  },
+          "responses": {
+            "200": { "type": "delivery" },
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "422":{ "type": "error" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/delivery.json:94]:
```suggestion
            "422": { "type": "error" }
```

**Review by Bhavya-verma-knoldus — APPROVED** [review]
lgtm

**Comment by jitendra2329** [comment]
Thank you for the Approval. will merge it into the main branch.
